File: mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/Python/Cubo/application/program.py
from entities.cubo import Cubo

def main():
    # Entrada dos lados dos cubos
    lado_x = float(input("Enter the side length of cubo X: "))
    lado_y = float(input("Enter the side length of cubo Y: "))

    # Criação dos objetos cubos
    cubo_x = Cubo(lado_x)
    cubo_y = Cubo(lado_y)

    # Cálculo dos volumes
    volume_x = cubo_x.calcular_volume()
    volume_y = cubo_y.calcular_volume()

    # Exibição dos volumes
    print(f"Volume of Cubo X = {volume_x}")
    print(f"Volume of Cubo Y = {volume_y}")

    # Comparação dos volumes
    if volume_x > volume_y:
        print("Larger volume: Cubo X")
    else:
        print("Larger volume: Cubo Y")

if __name__ == "__main__":
    main()

File: mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/Java/Cubo/src/application/Program.java
package application;

import entities.Cubo;
import java.util.Locale;
import java.util.Scanner;

public class Program {

    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        Cubo cuboX = new Cubo();
        Cubo cuboY = new Cubo();

        System.out.println("Enter the side length of cubo X: ");
        cuboX.lado = sc.nextDouble();

        System.out.println("Enter the side length of cubo Y: ");
        cuboY.lado = sc.nextDouble();

        double volumeX = cuboX.calcularVolume();
        System.out.println("Volume of Cubo X = " + volumeX);

        double volumeY = cuboY.calcularVolume();
        System.out.println("Volume of Cubo Y = " + volumeY);

        if (volumeX > volumeY) {
            System.out.println("Larger volume: Cubo X");
        } else {
            System.out.println("Larger volume: Cubo Y");
        }

        sc.close();
    }
}
